### ISSUE TITLE
Can not connect to VPN from inside CRC on Mac Laptop

### ISSUE THREAD

**Issue by jmesnil** [code-ready/crc#1299]
### General information

  * OS: macOS
  * Hypervisor: hyperkit
  * Did you run `crc setup` before starting it (Yes/No)? Yes
  * Running CRC on: Laptop

## CRC version
```bash
crc version: 1.11.0+883ca49
OpenShift version: 4.4.5 (embedded in binary)
```
  
## CRC status
```bash
CRC VM:          Running
OpenShift:       Running (v4.4.5)
Disk Usage:      13.64GB of 32.72GB (Inside the CRC VM)
Cache Usage:     24.07GB
Cache Directory: /Users/jmesnil/.crc/cache
```

## CRC config
```bash
- nameserver                            : 10.38.5.26
```

## Host Operating System
```bash
ProductName:    Mac OS X
ProductVersion: 10.15.5
BuildVersion:   19F101
```

### Steps to reproduce

I can not access machines on my company VPN from inside CRC.

I'm using OpenVPN to connect to my company VPN from my Mac laptop without any issue.
However inside CRC, I'm not able to connect to it and can not access our company internal image registry.

At first, I was not able to lookup the name server so I modified crc config to use the appropriate name server from inside CRC.
Now, the lookup works fine but I'm not able to ping the corresponding IP address.

  1. Connect to CRC VM: 

```
ssh -i ~/.crc/machines/crc/id_rsa -o StrictHostKeyChecking=no core@$(crc ip)
```

  2. Try to ping the machine inside the VPN:
```
[core@crc-m27h4-master-0 ~]$ ping registry-proxy.engineering.redhat.com
PING registry-proxy-prod.cloud.paas.psi.redhat.com (10.0.19.25) 56(84) bytes of data.

^C
--- registry-proxy-prod.cloud.paas.psi.redhat.com ping statistics ---
9 packets transmitted, 0 received, 100% packet loss, time 225ms
```

I tried a combination of steps to make it work to no avail:

* start CRC then connect to the VPN does not work
* connect to the VPN then start CRC does not work

I did not find any issue or doc that clearly explain how to setup CRC or VPN on Mac.
Do you have any idea to make it work please?

### Expected

I should be able to connect to machines behind my VPN from inside CRC.

### Actual

I am not able to connect to machines behind my VPN from inside CRC.

### Logs

[logs](https://gist.github.com/jmesnil/8894b7a9530ddcf6053597f95a6c15a9)

**Comment by gbraad** [comment]
> I should be able to connect to machines behind my VPN from inside CRC.

Why?

The host operating system RHCOS is supposed to be managed from the cluster itself and any access that requires access to another network might be best served with a proxy.

**Comment by gbraad** [comment]
if you mean to use a VPN besides the use of CRC you run into a known issue that is caused by the route all from the VPN. This is either configurable on your end, or if your admin prevents this change, it would be best to start CRC without the VPN connected as the local address would otherwise not be resolved/accessible.

**Comment by gbraad** [comment]
This has been documented in #384 and possible workarounds #1087 are still being worked on. Closing as duplicate of and known issue (see release notes).

**Comment by jmesnil** [comment]
@gbraad I was aware of #384 but turning off the VPN is not a workaround when our internal image registry is only available on the VPN...
The instructions on https://github.com/code-ready/crc/issues/384#issuecomment-546926921 apply to Linux and not Mac, I'll figure out how to make them work.

Will #1087 handle Windows only or will it it also handle Mac?

Thanks for the help...

**Comment by gbraad** [comment]
We would like to have feature parity between platforms, but need to target according to usage and timeline.

But you haven't answered the usage completely. you mention 'inside' the CRC VM as starting the VPN or resolving addresses. The latter is a configuration issue that is not related to CRC... and the first is not a usecase that is even described for RHCOS in general as I am aware of.

proxy-ing not possible?

**Comment by jmesnil** [comment]
sorry, I may have gone into more details before summarising my issue.

I want to run the command 
```
oc import-image jboss-eap-xp1-openjdk11-openshift:1.0 --from=registry-proxy.engineering.redhat.com/rh-osbs/jboss-eap-7-eap-xp1-openjdk11-openshift-rhel8:1.0 --insecure=true --confirm
``` 
where the `registry-proxy.engineering.redhat.com` machine is behind my company VPN.

The command fails with:

```
error: tag 1.0 failed: Internal error occurred: registry-proxy.engineering.redhat.com/rh-osbs/jboss-eap-7-eap-xp1-openjdk11-openshift-rhel8:1.0: Get http://registry-proxy.engineering.redhat.com/v2/: dial tcp: lookup registry-proxy.engineering.redhat.com on 172.30.0.10:53: no such host
imagestream.image.openshift.io/jboss-eap-xp1-openjdk11-openshift imported with errors
```

I'm able to pull the image without any issue from my Mac laptop directly:

```
docker pull registry-proxy.engineering.redhat.com/rh-osbs/jboss-eap-7-eap-xp1-openjdk11-openshift-rhel8:1.0
1.0: Pulling from rh-osbs/jboss-eap-7-eap-xp1-openjdk11-openshift-rhel8
Digest: sha256:94e1cd4eb4196a358e301c1992663258c0016c80247f507fd1c39cf9a73da833
Status: Image is up to date for registry-proxy.engineering.redhat.com/rh-osbs/jboss-eap-7-eap-xp1-openjdk11-openshift-rhel8:1.0
registry-proxy.engineering.redhat.com/rh-osbs/jboss-eap-7-eap-xp1-openjdk11-openshift-rhel8:1.0
```

When I started to investigate the issue, I realised that CRC VM is not able to connect to the VPN in the same way that my Mac does. Hence this bug.

Is my use case valid? Do you think that's an issue with OCP rather than crc (I should have asked that question first)?

Thanks

**Comment by gbraad** [comment]
> I'm able to pull the image without any issue from my Mac laptop directly:
> I realised that CRC VM is not able to connect to the VPN in the same way that my Mac does. 

It means that you are able to connect to the VPN without this interfering with your connectivity to the CRC VM. The problem you are describing is not directly related to the CRC< but rather the way the VM is rerouted to use the VPN network. Unfortunately there is no easy way to connect the VM to the VPN network using the hyperkit tools (non-existent tools). You should therefore have to look for a way to configure the VM to allow route to the VPN. I hope you understand this is not a CRC related issue, but rather infra and macOS. We would like there is an easy solution, so I suggest to use a proxy... you cna easily run a local proxy that acts as an intermediary between the VPN hosts and the VM.
